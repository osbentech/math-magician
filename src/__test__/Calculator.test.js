const { render, screen, fireEvent } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Calculator = require('../components/Calculator').default;

describe('Tests for Calculator component', () => {
  it('Renders Calculator page', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
  });

});