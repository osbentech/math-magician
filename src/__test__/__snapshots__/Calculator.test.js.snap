// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Calculator component Renders Calculator page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <header>
          <h1>
            A Benjamin think...
          </h1>
          <ul>
            <a
              href="/"
            >
              Home
            </a>
            <span>
               | 
            </span>
            <a
              href="/Calculator"
            >
              Calculator
            </a>
            <span>
               | 
            </span>
            <a
              href="/Quote"
            >
              Quotes
            </a>
          </ul>
        </header>
        <div
          class="caclContainer"
        >
          <h2>
            Hello, my name is Ben, let us do some maths..
          </h2>
          <div
            class="theCalculator"
          >
            <input
              class="editable"
              id="scr"
              value="0"
            />
            <div
              class="inputValues"
            >
              <div
                class="row1"
              >
                <button
                  class="AC"
                  type="button"
                >
                   AC
                </button>
                <button
                  class="plusNminus"
                  type="button"
                >
                  +/-
                </button>
                <button
                  class="percentage"
                  type="button"
                >
                  %
                </button>
                <button
                  class="division  arithmetic"
                  type="button"
                >
                  รท
                </button>
              </div>
              <div
                class="row2"
              >
                <button
                  class="seven"
                  type="button"
                >
                  7
                </button>
                <button
                  class="eight"
                  type="button"
                >
                  8
                </button>
                <button
                  class="nine"
                  type="button"
                >
                  9
                </button>
                <button
                  class="multiply  arithmetic"
                  type="button"
                >
                  x
                </button>
              </div>
              <div
                class="row3"
              >
                <button
                  class="four"
                  type="button"
                >
                  4
                </button>
                <button
                  class="five"
                  type="button"
                >
                  5
                </button>
                <button
                  class="six"
                  type="button"
                >
                  6
                </button>
                <button
                  class="minus  arithmetic"
                  type="button"
                >
                  -
                </button>
              </div>
              <div
                class="row4"
              >
                <button
                  class="one"
                  type="button"
                >
                  1
                </button>
                <button
                  class="two"
                  type="button"
                >
                  2
                </button>
                <button
                  class="three"
                  type="button"
                >
                  3
                </button>
                <button
                  class="plus  arithmetic"
                  type="button"
                >
                  +
                </button>
              </div>
              <div
                class="row5"
              >
                <button
                  class="zero"
                  type="button"
                >
                  0
                </button>
                <button
                  class="dot"
                  type="button"
                >
                  .
                </button>
                <button
                  class="equals  arithmetic"
                  type="button"
                >
                  =
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <header>
        <h1>
          A Benjamin think...
        </h1>
        <ul>
          <a
            href="/"
          >
            Home
          </a>
          <span>
             | 
          </span>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <span>
             | 
          </span>
          <a
            href="/Quote"
          >
            Quotes
          </a>
        </ul>
      </header>
      <div
        class="caclContainer"
      >
        <h2>
          Hello, my name is Ben, let us do some maths..
        </h2>
        <div
          class="theCalculator"
        >
          <input
            class="editable"
            id="scr"
            value="0"
          />
          <div
            class="inputValues"
          >
            <div
              class="row1"
            >
              <button
                class="AC"
                type="button"
              >
                 AC
              </button>
              <button
                class="plusNminus"
                type="button"
              >
                +/-
              </button>
              <button
                class="percentage"
                type="button"
              >
                %
              </button>
              <button
                class="division  arithmetic"
                type="button"
              >
                รท
              </button>
            </div>
            <div
              class="row2"
            >
              <button
                class="seven"
                type="button"
              >
                7
              </button>
              <button
                class="eight"
                type="button"
              >
                8
              </button>
              <button
                class="nine"
                type="button"
              >
                9
              </button>
              <button
                class="multiply  arithmetic"
                type="button"
              >
                x
              </button>
            </div>
            <div
              class="row3"
            >
              <button
                class="four"
                type="button"
              >
                4
              </button>
              <button
                class="five"
                type="button"
              >
                5
              </button>
              <button
                class="six"
                type="button"
              >
                6
              </button>
              <button
                class="minus  arithmetic"
                type="button"
              >
                -
              </button>
            </div>
            <div
              class="row4"
            >
              <button
                class="one"
                type="button"
              >
                1
              </button>
              <button
                class="two"
                type="button"
              >
                2
              </button>
              <button
                class="three"
                type="button"
              >
                3
              </button>
              <button
                class="plus  arithmetic"
                type="button"
              >
                +
              </button>
            </div>
            <div
              class="row5"
            >
              <button
                class="zero"
                type="button"
              >
                0
              </button>
              <button
                class="dot"
                type="button"
              >
                .
              </button>
              <button
                class="equals  arithmetic"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
